components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/RequestObjectServiceTest.java

```diff
@@ -0,0 +1,161 @@
+        List values = new ArrayList<>();
+        requestedEssentialClaims = new ArrayList<>();
+
+        requestedClaim.setName("email");
+        requestedClaim.setType("userinfo");
+        requestedClaim.setValue("value1");
+        requestedClaim.setEssential(true);
+        requestedClaim.setValues(values);
+        values.add("val1");
+        values.add("val2");
+        requestedClaim.setValues(values);
+        lstRequestedClams.add(requestedClaim);
+        requestedEssentialClaims.add(lstRequestedClams);
+    }
+
+    @Test
+    public void testAddRequestObject() throws Exception {
+
+        requestObjectService.addRequestObject(consumerKey, sessionKey, requestedEssentialClaims);
+        List<RequestedClaim> claims = requestObjectService.
+                getRequestedClaimsForSessionDataKey(sessionKey, true);
+        Assert.assertEquals(claims.get(0).getName(), "email");
+    }
+
+    @Test
+    public void testGetRequestedClaimsForUserInfo() throws Exception {
+
+        RequestObjectDAOImpl requestObjectDAO = new RequestObjectDAOImpl();
+        requestObjectService.addRequestObject(consumerKey, sessionKey, requestedEssentialClaims);
+        requestObjectDAO.updateRequestObjectReferencebyTokenId(sessionKey, tokenId);
+        addToken(token, tokenId);
+        List<RequestedClaim> claims = requestObjectService.getRequestedClaimsForUserInfo(token);
+        Assert.assertEquals(claims.get(0).getName(), "email");
+    }
+
+    @Test
+    public void testGetRequestedClaimsForUserInfoException() throws Exception {
+
+        String tableName = "IDN_OIDC_REQ_OBJECT_REFERENCE";
+        try {
+            RequestObjectDAOImpl requestObjectDAO = new RequestObjectDAOImpl();
+            requestObjectService.addRequestObject(consumerKey, sessionKey, requestedEssentialClaims);
+            requestObjectDAO.updateRequestObjectReferencebyTokenId(sessionKey, invalidTokenId);
+            addToken(token, tokenId);
+            List<RequestedClaim> claims = requestObjectService.getRequestedClaimsForUserInfo(token);
+            Assert.assertEquals(claims.get(0).getName(), "email");
+        } catch (Exception e) {
+            Assert.assertEquals(e.getMessage(), "Can not update code id or the access token id of the " +
+                    "table ." + tableName);
+        }
+    }
+
+    @Test
+    public void testGetRequestedClaimsForIDToken() throws Exception {
+
+        RequestObjectDAOImpl requestObjectDAO = new RequestObjectDAOImpl();
+        requestObjectService.addRequestObject(consumerKey, sessionKey, requestedEssentialClaims);
+        requestObjectDAO.updateRequestObjectReferencebyTokenId(sessionKey, tokenId);
+        updateUserInfo(sessionKey, false);
+        addToken(token, tokenId);
+        List<RequestedClaim> claims = requestObjectService.getRequestedClaimsForIDToken(token);
+        Assert.assertEquals(claims.get(0).getName(), "email");
+    }
+
+    protected void addToken(String token, String tokenId) throws Exception {
```

Comment:
instead of calling this method, you cann use the DAO class . check AccessTokenDAOImpl

Reply:
Addressed with 5a21a406f90e20e60b84b767c04f5271f9f599bd